packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -1119,6 +1117,14 @@ Future<DateTimeRange?> showDateRangePicker({
     initialDateRange == null || !initialDateRange.end.isAfter(lastDate),
     "initialDateRange's end date must be on or before lastDate $lastDate.",
   );
+  assert(
```

Comment:
Is this assertion always true ? You mentioned an example with an hotel that might be unavailable. Here is the scenario I have in mind:

An user on an application selects a date range, let's say from monday to friday of the first week of a given month. After selection, the hotel becomes unavailable on friday. The frontend application listens to changes in the hotel availability and wants to display that the current selection is now unavailable ( it wasn't before)

Reply:
So maybe we should unselect them?

Reply:
This is an edge case, I think it is fine how it is now. It's on the application scope to unselect if that's the case.

Reply:
Ok so we can merge it now?

Reply:
I believe there are some formatting issues that you may want to fix before @MitchellGoodwin review

Reply:
Which ones ? I tried my best regarding the docs, and as said inside them, it's "manually" formatted 😄 Tell me the needed changes

Reply:
There is some longer lines, but as said in the docs, it's accepted if it improves the reading

Reply:
You might wanna ping @MitchellGoodwin on discord for the review, as described in the wiki

Reply:
Oh I missed this part, I'll read again. Thanks for the info !